src/compose/ComposeBox.js

```diff
@@ -441,6 +546,68 @@ export default function ComposeBox(props: Props): Node {
     messageInputState,
   ]);
 
+  const doQuoteAndReply = useCallback(
+    async message => {
+      const user = allUsersById.get(message.sender_id);
+      if (!user) {
+        showErrorAlert(_('Quote-and-reply failed'));
+        logging.error('Missing user for sender_id in quote-and-reply', {
+          sender_id: message.sender_id,
+        });
+        return;
+      }
+
+      // TODO: If not already there, re-narrow to `message`'s conversation
+      //   narrow, with getNarrowForReply, and do the quote-and-reply there.
+      //   Discussion:
+      //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M1975.20Quote.20and.20reply/near/1455302
+
+      setQuoteAndReplyInProgress(true);
```

Comment:
Is there anything in this version to prevent having two quote-and-reply actions happening at the same time? I think there isn't.

If two (or more) of these happen at the same time, that'll be bad news:
* They both/all use the same placeholder text -- so whichever finishes first will replace the first occurrence of the placeholder.
* When one finishes, it'll set `quoteAndReplyInProgress` all the way to false, which will re-enable sending even though the others are still working.

A simple, adequate fix would be to… hmm, I was going to say here this function could just give an error up front if there's a quote-and-reply in progress. But because `quoteAndReplyInProgress` is stored as React state, I think that would actually still leave a race: invoking the function twice in quick succession, so that `ComposeBox` didn't re-render between the invocations, would still act twice.

So one correct fix would be to use a ref for `quoteAndReplyInProgress`, rather than React state. That means this function would always see the correct answer as to whether such an action is actually in progress, rather than whether the UI has updated to reflect that one is. … A bit sticky how to handle that in `validationErrors`, though, which is used in rendering the UI. The flag could be duplicated as *both* a ref and a piece of React state? I think that would work right, but it'd be getting messy.

The alternate path would be to just handle the general case: allow a bunch of quote-and-reply actions to happen concurrently. That means:
* Keep a count of invocations, like `insertAttachment` does, rather than a boolean, so that we can decrement rather than setting to false.
* Use distinct placeholder strings. The first one can still be plain "[Quoting…]", but then after that something like "[Quoting (1)…]", "[Quoting (2)…]", etc.
  * The counter to drive these needs to be in a ref, not React state.